pyxlsb/styles.py

```diff
@@ -25,14 +67,47 @@ def _parse(self):
         self._cell_styles = list()
         self._cell_style_xfs = list()
 
+        self._xf_record = dict()
+        self._format_record = dict()
+
         self._fp.seek(0, os.SEEK_SET)
         for rectype, rec in RecordReader(self._fp):
             # TODO
-            if rectype == rt.END_STYLE_SHEET:
+            if rectype == rt.XF:
+                self._xf_record[len(self._xf_record) - 1] = rec
```

Comment:
This here should be using the `rec.xfId` as the key: `[rec.xfId]`

Reply:
Now `rec.xfId` is always zero. So if we use `self._xf_record[rec.xfId] = rec` it will not be right.
But if you want fill value as 
```
rec.xfId = len(self._xf_record) - 1
self._xf_record[rec.xfId] = rec
```
than ok.

Reply:
Ok so I went back to MS-XLSB, if you read section 3.6 they have an example for styling. Based on that, what you're doing here is only a portion of the required processing. Also my initial draft for the style structure aren't quite good enough.

Turns out that the "ID" of the style really is it's index in the structure, you'll have something like this in the binary styles part:

```
BrtBeginCellStyleXFs
  BrtXF
BrtEndCellStyleXFs
BrtBeginCellXFs
  BrtXF
  BrtXF
  ...
BrtEndCellXFs
```

Like I originally said, both `BrtBeginCellStyleXFs` and `BrtBeginCellXFs` will contain the count of records, this allows you to use lists because you will know ahead of time which size they will need. Also, since "ID" are actually indices, with lists it's a natural key.

`BrtXF` records under `BrtBeginCellStyleXFs` must have an `ixfeParent` set to `0xFFFF`while the record under `BrtBeginCellXFs` must have a reference to a record in the former structure. The `BrtCell*` records must point to a `CellXF` record which then must points to a `CellStyleXF`. The `xfGrbitAtr` bitfield specifies which property of the `CellStyleXF` is overwritten by the `CellXF` (this could include the format ID), section 2.4.844 specifies that a high bit means "overwrite".

What this all means is that you'll have to progressively parse both collections, switching when you encounter the boundaries. Then we'll have to at least interpret the overwrite for the format ID in the cell XF and resolve it's parent XF if needed.

BTW, you can get MS-XLSB here: https://docs.microsoft.com/en-us/openspecs/office_file_formats/ms-xlsb/acc8aa92-1f02-4167-99f5-84f9f676b95a

Reply:
As I understand It could be speculative and seems wrong, but I test on my files and now it works good. Now we have two different lists: where xfId is None and not None.

Reply:
I guess this will work, at least partially. It'll do for now, we can improve later.